cache_flat_mutation_reader.hh

```diff
@@ -620,6 +620,11 @@ void cache_flat_mutation_reader::add_to_buffer(const partition_snapshot_row_curs
         _read_context->cache().on_row_hit();
         add_clustering_row_to_buffer(mutation_fragment(*_schema, _permit, row.row(_read_context->digest_requested())));
     } else {
+        position_in_partition::less_compare less(*_schema);
+        if (less(_lower_bound, row.position())) {
```

Comment:
When can this be false?

Reply:
For example, when the query range starts with a position which is equal to dummy entry's position.